service/share/cache_availability.go

```diff
@@ -50,18 +53,22 @@ func (ca *CacheAvailability) SharesAvailable(ctx context.Context, root *Root) er
 	}
 	// do not sample over Root that has already been sampled
 	key := rootKey(root)
+
+	ca.dsLk.Lock()
 	exists, err := ca.ds.Has(ctx, key)
-	if err != nil {
+	ca.dsLk.Unlock()
+	if err != nil || exists {
 		return err
 	}
-	if exists {
-		return nil
-	}
+
 	err = ca.avail.SharesAvailable(ctx, root)
 	if err != nil {
 		return err
 	}
+
+	ca.dsLk.Lock()
 	err = ca.ds.Put(ctx, key, []byte{})
+	ca.dsLk.Unlock()
```

Comment:
By contention point TODO I meant specifically this place. Basically, the routine hitting the autobatching threshold will lock all the other writers in future parallelized DASer until the batch is synced on disk. This fundamentally degrades the parallelization of a DASer. The solution I know is to decouple reads and write in header.Store

Reply:
Can you document this pls on #483

Reply:
whoops accidentally resolved.. I don't know if u want to resolve or not